src/sentry/notifications/utils/participants.py

```diff
@@ -351,21 +353,59 @@ def get_fallthrough_recipients(
     ):
         return []
 
+    def get_all_members() -> Iterable[APIUser]:
+        user_ids = project.member_set.all().values_list("user_id", flat=True)
+        return list(
+            UserService.serialize_user(user) for user in User.objects.filter(id__in=user_ids)
+        )
+
     # Case 1: No fallthrough
     if fallthrough_choice == FallthroughChoiceType.NO_ONE:
         return []
 
     # Case 2: notify all members
     elif fallthrough_choice == FallthroughChoiceType.ALL_MEMBERS:
-        user_ids = project.member_set.all().values_list("user_id", flat=True)
-        return list(
-            UserService.serialize_user(user) for user in User.objects.filter(id__in=user_ids)
-        )
+        return get_all_members()
 
-    # TODO(snigdhasharma): Handle this in a followup PR (WOR-2385)
     # Case 3: Admin or recent
     elif fallthrough_choice == FallthroughChoiceType.ADMIN_OR_RECENT:
-        return []
+        if len(project.member_set) < FALLTHROUGH_NOTIFICATION_LIMIT:
+            return get_all_members()
+
+        # We notify at most FALLTHROUGH_NOTIFICATION_LIMIT people, so we'll try to notify all admins first
+        team_admins = (
+            OrganizationMember.objects.get_contactable_members_for_org(project.organization_id)
+            .filter(
+                teams__in=project.teams.all(),
+                role__in=(r.id for r in roles.get_all() if r.has_scope("project:admin")),
+            )
+            .values_list("user_id", flat=True)
+        )
+
+        actor_ids = []
+        if len(team_admins) > FALLTHROUGH_NOTIFICATION_LIMIT:
+            actor_ids = team_admins[:FALLTHROUGH_NOTIFICATION_LIMIT]
```

Comment:
Why do we cap at 20? The 21st Admin is going to be the happiest :)

Reply:
LOL I think we said we wanted to only notify 20 folks, not sure if that applied to admins as well. I'll check with Dhrumil